src/api/services/ingestion.go

```diff
@@ -274,28 +274,69 @@ func (i *IngestionService) UploadVcfGzToDrs(cfg *models.Config, drsBridgeDirecto
 	data := fmt.Sprintf("{\"path\": \"%s/%s\"}", drsBridgeDirectory, gzippedFileName)
 
 	var (
-		drsResp *http.Response
-		drsErr  error
+		drsId           string
+		drsResp         *http.Response
+		drsErr          error
+		attemptCount    int = 0
+		maxAttempts     int = 5
+		waitTimeSeconds int = 3
 	)
 	for {
+		// prepare upload request to drs
 		r, _ := http.NewRequest("POST", drsUrl+"/private/ingest", bytes.NewBufferString(data))
-		r.SetBasicAuth(drsUsername, drsPassword)
 
+		r.SetBasicAuth(drsUsername, drsPassword)
 		r.Header.Add("Content-Type", "application/json")
+
 		client := &http.Client{}
+
+		// perform request
 		drsResp, drsErr = client.Do(r)
+
+		// check for errors, possibly try again
 		if drsErr != nil {
 			fmt.Printf("Upload to DRS error: %s\n", drsErr)
-			return ""
-		} else {
-			fmt.Printf("Upload to DRS succeeded: %d\n", drsResp.StatusCode)
+
+			if attemptCount < maxAttempts {
+				// increment attempt counter
+				attemptCount++
+
+				// give it a few seconds break
+				time.Sleep(time.Duration(waitTimeSeconds * int(time.Second)))
+
+				fmt.Printf("trying again...\n")
+				continue
+			} else {
+				fmt.Printf("exiting upload loop...\n")
+				return "" // empty drs-id string
+			}
 		}
 
 		// check for simple upload error (like db locked) and try again
-		if drsResp.StatusCode != 400 {
+		fmt.Printf("Got a %d status code on DRS upload \n", drsResp.StatusCode)
+		if drsResp.StatusCode == 201 {
+			fmt.Printf("File %s upload to DRS succeeded: %d\n", gzippedFileName, drsResp.StatusCode)
+
+			// proceed with vcf processing
 			break
+		} else if drsResp.StatusCode == 401 {
+			// exit right away on 'unauthorized' status code
+			fmt.Printf("exiting upload loop...\n")
```

Comment:
To help debugging, consider providing a message that reports an error wrt authorization